components/widgets/ui/src/main/resources/PhenoTips/GeneVariantMacros.xml

```diff
@@ -0,0 +1,1420 @@
+      $$('.variant-gene-GENE_INDEX_PLACEHOLDER').each(function(item) {
+        item.remove();
+      });
+    },
+    ajaxDeleteGeneData : function (event) {
+      event.stop();
+      var deleteTrigger = event.element();
+      var className = deleteTrigger.up('td').className;
+      var geneIndex = parseInt(className.substring(className.lastIndexOf('-')+1));
+      var geneSymbol = $$('[name="'+this.geneClassName+'_' + geneIndex + '_gene"]')[0].value;
+      new XWiki.widgets.ConfirmedAjaxRequest(deleteTrigger.href+geneSymbol, {
+        onCreate : function() {
+          deleteTrigger.disabled = true
+        },
+        onSuccess : function() {
+          var dataRow = deleteTrigger.up('tr:not(.head-group)');
+          var dataTable = deleteTrigger.up('table.gene-table.extradata-list');
+          if (dataRow) {
+            $$('.variant-gene-'+geneIndex).each(function(item) {
+              item.remove();
+            });
+            dataRow.remove();
+          }
+          if (dataTable) {
+            var i = 1;
+            dataTable.select('td.row-count').each(function(item) {
+              var geneRowIndex = item.next().className.substring(item.next().className.lastIndexOf('-')+1);
+              var y = 1;
+              $$('.variant-hide-heading-' + geneRowIndex + ' .variant-row-count').each(function(vitem) {
+                vitem.update(i + '.' + (y++));
+              });
+              item.update(i++);
+            });
+          }
+        },
+        onComplete : function() {
+          deleteTrigger.disabled = false;
+        }
+      },
+      {
+         confirmationText : "$!services.localization.render('phenotips.tableMacros.rowDeleteConfirmation')"
+      });
+    },
+    ajaxAddGeneData : function (event) {
+      event.stop();
+      var addTrigger = event.element();
+      if (addTrigger.disabled) {
+        return;
+      }
+      
+      var dataTable = addTrigger.up('.list-actions').previous();
+      if (!dataTable) {
+        new XWiki.widgets.Notification("$services.localization.render('phenotips.tableMacros.listNotFound')", 'error');
+      }
+      
+      var geneRowTemplateEl = this.newGeneRowElement;
+      var buttonRowTemplateEl = this.addVariantButtonElement;
+
+      var idx = dataTable.select('td.row-count').size() + 1;
+      var geneIndex = 0;
+      if (dataTable.select('td.row-count').size() > 0) {
+        var className = dataTable.select('td.row-count')[idx - 2].next().className;
+        geneIndex = parseInt(className.substring(className.lastIndexOf('-')+1)) + 1;
+      }
+
+      var newGeneRow = new Element('tr');
+      var newButtonRow = new Element('tr', {'class' : 'variant-gene-'+geneIndex+' variant-footer v-collapsed'}); 
+
+      var geneRowInner = geneRowTemplateEl.innerHTML.replace( /GENE_NUMBER_PLACEHOLDER/g, idx).replace( /GENE_INDEX_PLACEHOLDER/g, geneIndex).replace(/_\d+_/g, '_'+geneIndex+'_').replace(/=\d+&amp;/g, '='+geneIndex+'&amp;');
```

Comment:
prototypes doesnt seem to work, however, the .replace() itself surprizingly can do a templating, will stick to that solution:
var res = str.replace(/blue|house|car/gi, function myFunction(x){return x.toUpperCase();});